MLS.Agent/NativeAssemblyLoadHelper.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using static WorkspaceServer.Kernel.CSharpKernelExtensions;
+
+namespace MLS.Agent
+{
+    public class NativeAssemblyLoadHelper : INativeAssemblyLoadHelper
+    {
+        private readonly string _tfm;
+
+        public NativeAssemblyLoadHelper()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                _tfm = "osx-x64";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                _tfm = "linux-x64";
+            }
+            else if (RuntimeInformation.OSArchitecture == Architecture.X86)
+            {
+                _tfm = "win-x86";
+            }
+            else
+            {
+                _tfm = "win-x64";
+            }
+        }
+
+        public void Handle(string assembly)
+        {
+            AppDomain currentDomain = AppDomain.CurrentDomain;
+            currentDomain.AssemblyLoad += AssemblyLoaded(assembly);
+        }
+
+        private AssemblyLoadEventHandler AssemblyLoaded(string assembly)
+        {
+            return (object sender, AssemblyLoadEventArgs args) =>
+            {
+                if (args.LoadedAssembly.Location == assembly)
+                {
+                    NativeLibrary.SetDllImportResolver(args.LoadedAssembly, Resolve);
+                }
+            };
+        }
+
+        private IntPtr Resolve(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            var basePath = Path.GetDirectoryName(assembly.Location);
+            var nativeAssembly = Path.Combine(basePath, "..", "..", "runtimes", _tfm, "native", libraryName) + ".dll";
```

Comment:
This should really be using the NuGet native assets in the `project.assets.json`, and not assuming this path.

For one thing, this isn't taking into account the RID-fallback. You can have native assets in a `debian-x64` folder, and those will be used on any debian based system. They don't have to be just in `linux-x64`.

Reply:
Or even better would be the `.deps.json` file, but I assume you aren't getting one since you aren't going through the full SDK/MSBuild process.

Reply:
Yes, the plan is to parse deps.json, this is basically some demoware based on partner requests :)